app/utils/validation.ts

```diff
@@ -7,61 +7,53 @@ export class InvalidUsernameError extends Error {
   }
 }
 
-export class UsernameValidation {
-  static usernameRegex = /(?!^(1|3|bc1|lnbc1))^[0-9a-z_]{3,50}$/i
-
-  public static Invalid3CharactersMin = translate("UsernameScreen.3CharactersMinimum")
-  public static Invalid50CharactersMax = translate("UsernameScreen.50CharactersMaximum")
-  public static InvalidLettersAndNumbers = translate("UsernameScreen.letterAndNumber")
-  public static InvalidEmailAddress =
-    translate("UsernameScreen.emailAddress") +
-    ". " +
-    translate("UsernameScreen.letterAndNumber")
-  public static InvalidStartCharacters = translate("UsernameScreen.forbiddenStart")
-  public static InvalidUnexpected = translate("errors.unexpectedError")
-
-  public static checkedToUsername = (username: string): string | InvalidUsernameError => {
-    if (!username.match(this.usernameRegex)) {
-      switch (false) {
-        case this.hasValidLength(username):
-          return username.length > 3
-            ? new InvalidUsernameError(this.Invalid50CharactersMax)
-            : new InvalidUsernameError(this.Invalid3CharactersMin)
-        case this.hasValidCharacters(username):
-          if (this.isEmailAddress(username)) {
-            return new InvalidUsernameError(this.InvalidEmailAddress)
-          }
-          return new InvalidUsernameError(this.InvalidLettersAndNumbers)
-        case this.hasNoRestictedStartCharacters(username):
-          return new InvalidUsernameError(this.InvalidStartCharacters)
-        default:
-          return new InvalidUsernameError(this.InvalidUnexpected)
-      }
+export const validateUsername = (username: string) => {
+  const usernameRegex = /(?!^(1|3|bc1|lnbc1))^[0-9a-z_]{3,50}$/i
+
+  if (!username.match(usernameRegex)) {
+    switch (false) {
```

Comment:
`switch(false)`? what is the goal of this statement?

Reply:
To be clear I didn't add this code I was just moving it into a function.  Looks like the switch is looking for cases which return false (i.e. fail validation in some way).